### DISCUSSION TITLE
How to create mutable objects?

### DISCUSSION THREAD

**Discussion by Shreehar01** [nasa/openmct#5210] [Community Q&A]
The openmct API documentation mentions that "object providers will support other methods to enable other operations with persistence stores, such as creating, updating, and deleting objects." Is there any other way, for now, we can create a mutable object?

I am seeing the error:
`Uncaught (in promise) Error: Attempted to mutate immutable object Overview openmct.js:329:582425`

The use of GetMutable() for my object from the object provider instead of just Get() also throws an error:
`Uncaught (in promise) Error: Object "tbc.loop:vehicles" does not support mutation.`

When I look into their code for the Object in the github, it decides if an object is mutable or immutable based on if it has create and update functions. However, in the absence of create and update functions for the time being (as mentioned above and in the documentation), how can I create a mutable object to eliminate the errors I am receiving?

**Comment by akhenry** [comment]
@Shreehar01 Can you tell us a little bit more about your use case? Mutable objects have to have some way of persisting any mutation that occurs (ie via a corresponding `update` function on a provider). If you are attempting to edit and save an immutable object then this error message is appropriate, however if you are seeing this error message outside of an editing workflow then this might be a bug in Open MCT.

**Comment by akhenry** [comment]
@akhenry
